### PR TITLE
Initial proof of concept

### PR DESCRIPTION
**Pull request by Kreest**
Sending annotations works, testing it on my reader.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11248)

### CODE REVIEW

**Comment by Kreest** [comment]
whoops sorry, wanted this to go on my clone first

**Comment by Frenzie** [comment]
np, looks interesting :+1:

File: plugins/wallabag.koplugin/main.lua [hunk 1]
```diff
@@ -883,11 +905,113 @@ function Wallabag:addTags(path)
...
+            false
+end
+
+function Wallabag:sendAnnotations(filepath)
+    if self.ui.doc_settings then
+        self.ui.doc_settings:flush()
+    end
+    logger.dbg("Wallabag: sending annotations for article ", filepath)
+    local document = DocumentRegistry:openDocument(filepath)
+    document:loadDocument() -- wait is this done async, do we need to wait for this?
```
[the remark this answers is not in the document]
Comment (Kreest) [reply plugins/wallabag.koplugin/main.lua:920]: The two annotation formats are just a tiny bit different enough that I don't believe loading the document can be avoided.

The koreader format subdivides the text on child elements of the xml paragraph (such as the italicized _not_ below) while the wallabag format inherited from [here](https://github.com/openannotation/xpath-range) just counts letters from the start of the paragraph. Would there be a way to get that count without loading the document?

**Example**:
For this note:
2023-12-21-165645_grim

The koreader format for the position looks like this:
```
["pos0"] = "/body/DocFragment[3]/body/p[2]/text()[2].1",
["pos1"] = "/body/DocFragment[3]/body/p[2]/text()[2].32",
```

And the same note's position in wallabag format looks like this:
```
"ranges": [
	{
		"start": "\/p[2]",
		"startOffset": "317",
		"end": "\/p[2]",
		"endOffset": "406"
	}
]
```

[the remark this answers is not in the document]
Comment (Frenzie) [reply plugins/wallabag.koplugin/main.lua:920]: No, I don't think that would be possible.

[the remark this answers is not in the document]
Comment (Frenzie) [reply plugins/wallabag.koplugin/main.lua:920]: With the minor sidenote that, if enabled, the Wallabag plugin could perhaps do so during the highlight or saving process if the file is in the Wallabag folder.

**Pull request closed without merging**